blocks/core.py

```diff
@@ -20,22 +20,24 @@
     Union,
 )
 
+from blocks.filesystem import FileSystem
+from blocks.utils import with_function_tmpdir, with_session_tmpdir
 from blocks.dfio import read_df, write_df
-from blocks.filesystem import GCSFileSystem, GCSNativeFileSystem, FileSystem
-from blocks.datafile import DataFile
 
 cgroup = str
 rgroup = str
 
 
+@with_function_tmpdir
 def assemble(
     path: str,
     cgroups: Optional[Sequence[cgroup]] = None,
     rgroups: Optional[Sequence[rgroup]] = None,
     read_args: Any = {},
     cgroup_args: Dict[cgroup, Any] = {},
     merge: str = "inner",
-    filesystem: FileSystem = GCSFileSystem(),
+    filesystem: FileSystem = FileSystem(),
+    tmpdir: str = None,
```

Comment:
nit: Optional[str]

Reply:
yeah i'm confused what to do here with typehints/docstring, this is not actually optional at all but is provided by the decorator instead so the user should never pass it in